src/client/hud.cpp

```diff
@@ -862,6 +862,56 @@ void Hud::drawSelectionMesh()
 	}
 }
 
+void Hud::toggleBlockBounds()
+{
+	m_block_bounds_mode = static_cast<BlockBoundsMode>(m_block_bounds_mode + 1);
+
+	if (m_block_bounds_mode >= BLOCK_BOUNDS_MAX) {
+		m_block_bounds_mode = BLOCK_BOUNDS_OFF;
+	}
+}
+
+void Hud::drawBlockBounds()
+{
+	if (m_block_bounds_mode == BLOCK_BOUNDS_OFF) {
+		return;	
+	}
+
+	video::SMaterial old_material = driver->getMaterial2D();
+	driver->setMaterial(m_selection_material);
+
+	v3s16 pos = player->getStandingNodePos();
+
+	v3s16 blockPos(
```

Comment:
`v3s16 blockPos = pos / MAP_BLOCKSIZE;` should work equally well
(int divisions behaves identically to flooring)

Reply:
Well yes they don't. https://github.com/minetest/minetest/pull/11172#discussion_r611200013

Reply:
Ah of course, negative numbers exist.